src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -80,6 +83,95 @@ enum MiscSpells
     SPELL_MAGE_CHILLED                           = 205708
 };
 
+// 110909 - Alter Time
+class spell_mage_alter_time_aura : public AuraScript
+{
+    PrepareAuraScript(spell_mage_alter_time_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_MAGE_ALTER_TIME,
+            SPELL_MAGE_ALTER_TIME_AURA,
+            SPELL_MAGE_ALTER_TIME_ACTIVE,
+            });
+    }
+
+    void Apply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* unit = GetCaster();
+        if (unit == nullptr)
+            return;
+
+        if (Player* player = unit->ToPlayer())
+        {
+            _health = player->GetHealth();
+            _loc = player->GetWorldLocation();
+        }
+    }
+
+    void Remove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* unit = GetCaster();
```

Comment:
Wrong choice of function to use here

`GetTarget()` - the unit with AuraApplication, should be your default choice in scripts
`GetCaster()` - unit that initially cast the spell
`GetUnitOwner()` - unit that "owns" the aura, use only for AoE auras when you need to access the unit that is the source of all AuraApplications